File: Leetcode September Daily Challenge/Leetcode #948/C++/sol.cpp
class Solution {
public:
    int bagOfTokensScore(vector<int>& tokens, int power) {
        int score = 0,l=0,r=tokens.size()-1;
        sort(tokens.begin(),tokens.end());
        while(l<=r)
        {
            if(l <=r && power - tokens[l] >=0)
            {
                power-=tokens[l];
                l++;
                score++;
            }
            else if(l<r && score > 0 && power-tokens[l] <0 )
            {
                power+= tokens[r];
                r--;
                score--;
            }
            else break;
            
        }
        return score;
    }
};

File: Leetcode September Daily Challenge/Leetcode #948/Python/sol.py
class Solution:
    def bagOfTokensScore(self, tokens: List[int], power: int) -> int:
        
        if not tokens or power<min(tokens):
            return 0
        i=0
        j=len(tokens)-1
        tokens.sort()
        score=0
        while i<=j:
            if tokens[i]<=power:
                score+=1
                power-=tokens[i]
                i+=1
            elif i==j:
                break
            else:
                score-=1
                power+=tokens[j]
                j-=1
        return score
